### PR TITLE
Fix spectrum plots for different states

### PR DESCRIPTION
**Pull request by iaraota**
This PR addresses a bug related to computing the _spectrum_ for different states. Previously, the same spectrogram was being plotted for different states despite having different segments, due to a global variable storing spectrum data with identical keys for all states. This issue has been resolved by introducing the `state` argument to the `get_spectrum` and `get_spectrum_range` functions, using the state name as a key to differentiate spectrum data.

The original intention behind the removed snippet below

```python
if isinstance(segments, DataQualityFlag):	
    name = ','.join([channel.ndsname, segments.name])
    segments = segments.active
else:
    name = channel.ndsname
```
was to incorporate the state into the `name` variable. However, this condition always failed because of the decorator `use_segmentlist`, which was already reseting `segments = segments.active`, therefore making it impossible to insert the state in the `name` variable. With the introduction of the `state` variable, this redundant condition has been removed.

Additionally, the range _spectrogram_ also required fixing as it failed to change with the state. The code introduced mirrors that used in `_get_spectrogram` to ensure the correct data is returned.

## Example:
- h(t) spectrum: https://ldas-jobs.ligo-la.caltech.edu/~iara.ota/summary/day/20240322/lock/strain
- range spectrum and spectrogram: https://ldas-jobs.ligo-la.caltech.edu/~iara.ota/summary/day/20240322/lock/range/
- Network seimic noisy and quiet states: https://ldas-jobs.ligo.caltech.edu/~iara.ota/summary/day/20240325/sei/

### CODE REVIEW

File: gwsumm/data/range.py [hunk 1]
```diff
@@ -117,21 +117,35 @@ def get_range_spectrogram(channel, segments, config=None, cache=None,
...
+                    out.append(s)
+    return out.coalesce()
 
 
 @use_segmentlist
 def get_range_spectrum(channel, segments, config=None, cache=None, query=True,
                        nds=None, return_=True, nproc=1, datafind_error='raise',
                        frametype=None, stride=60, fftlength=None, overlap=None,
-                       method=None, which='all', **rangekwargs):
+                       method=None, which='all', state='', **rangekwargs):
```
Comment (eagoetz) [inline gwsumm/data/range.py:142]: I prefer not having an empty string be the "no state given" value. I suggest using `None` and then I'll suggest another change on L146 assuming you're fine this suggestion
```suggestion
                       method=None, which='all', state=None, **rangekwargs):
```

File: gwsumm/data/range.py [hunk 2]
```diff
@@ -117,21 +117,35 @@ def get_range_spectrogram(channel, segments, config=None, cache=None,
...
     """Compute percentile spectra of the range integrand from a set of
     spectrograms
     """
-    name = str(channel)
-    cmin = '%s.min' % name
-    cmax = '%s.max' % name
+    name = str(channel) + f',{state}'
```
Comment (eagoetz) [inline gwsumm/data/range.py:146]:
```suggestion
    name = str(channel)
    if state:
        name += f',{state}'
```

File: gwsumm/data/spectral.py [hunk 3]
```diff
@@ -328,17 +327,11 @@ def apply_transfer_function_series(specgram, tfunc):
 def get_spectrum(channel, segments, config=None, cache=None,
                  query=True, nds=None, format='power', return_=True,
                  frametype=None, nproc=1, datafind_error='raise',
-                 **fftparams):
-    """Retrieve the time-series and generate a spectrogram of the given
+                 state='', **fftparams):
```
Comment (eagoetz) [inline gwsumm/data/spectral.py:330]: Same as above
```suggestion
                 state=None, **fftparams):
```

File: gwsumm/data/spectral.py [hunk 4]
```diff
@@ -363,19 +357,15 @@ def get_spectrum(channel, segments, config=None, cache=None,
 
 def _get_spectrum(channel, segments, config=None, cache=None, query=True,
                   nds=None, format='power', return_=True, which='all',
-                  **fftparams):
-    """Retrieve the time-series and generate a spectrogram of the given
+                  state='', **fftparams):
```
Comment (eagoetz) [inline gwsumm/data/spectral.py:360]:
```suggestion
                  state=None, **fftparams):
```

File: gwsumm/data/spectral.py [hunk 5]
```diff
@@ -363,19 +357,15 @@ def get_spectrum(channel, segments, config=None, cache=None,
...
-    if isinstance(segments, DataQualityFlag):
-        name = ','.join([channel.ndsname, segments.name])
-        segments = segments.active
-    else:
-        name = channel.ndsname
-    name += ',%s' % format
-    cmin = '%s.min' % name
-    cmax = '%s.max' % name
+
+    name = f'{channel.ndsname},{state},{format}'
```
Comment (eagoetz) [inline gwsumm/data/spectral.py:366]:
```suggestion
    name = f'{channel.ndsname},{format}'
    if state:
        name = f'{channel.ndsname},{state},{format}'
```

File: gwsumm/data/spectral.py
Comment (eagoetz) [inline gwsumm/data/spectral.py:1]: Do you also need to use the `state` argument in the call to `get_spectrogram` on L381-3?

Comment (eagoetz) [reply gwsumm/data/spectral.py:1]: Please ignore, I think I was confused here

**Comment by eagoetz** [comment]
@iaraota Thanks for finding the source of this bug! I think that we also need to fix the spectrograms, correct? For example, check out quiet and noisy state spectrograms here: https://ldas-jobs.ligo.caltech.edu/~detchar/summary/day/20240101/sei/bs/

**Comment by iaraota** [comment]
> @iaraota Thanks for finding the source of this bug! I think that we also need to fix the spectrograms, correct? For example, check out quiet and noisy state spectrograms here: https://ldas-jobs.ligo.caltech.edu/~detchar/summary/day/20240101/sei/bs/

I didn't realize that, as the spectrogram for the strain works fine. I will have a look, thank you!

**Comment by iaraota** [comment]
@eagoetz I was looking into the problem of [quiet and noise spectrograms](https://ldas-jobs.ligo.caltech.edu/~detchar/summary/day/20240101/sei/bs/) and it seem to me that the problem is having a different definition for H1 and L1, in the configuration file we have:

```ìni
[state-Quiet]
key = Quiet
name = SEI Quiet

[state-H1-Quiet]
name = SEI Quiet
key = H1-quiet
hours = 1-5,H1

[state-L1-Quiet]
name = SEI Quiet
key = L1-quiet
hours = 2-6,L1
```

And this page uses the state `Quiet` and intersect the options for L1 and H1, and that is causing the problem. For example, when using the state `H1-quiet/noisy`, the issue doesn't arise:  https://ldas-jobs.ligo.caltech.edu/~iara.ota/summary/day/20240404/sei/bs

Thus, the issue does not lie within the `get_spectrogram` function, but rather in how the states are generated, considering the discrepancies between the two sites. In my view, this problem falls outside the scope of this PR and should be addressed separately in another PR.

**Comment by eagoetz** [comment]
@iaraota Yeah, I had a feeling it was more complicated, but at least we have some fixes for the spectra. Let's go ahead with this PR and continue working on the spectrogram issue in another PR

**Review by eagoetz — APPROVED** [review]
Thanks again @iaraota! LGTM

**Pull request merged**